Monika After Story/game/event-handler.rpy

```diff
@@ -1628,6 +1623,78 @@ init python:
         # otherwise we didnt unlock anything because nothing available
         return False
 
+#TODO:Need to block talk, extra and play from being pressed while eventlabel inactivity_detection is being called/jumped to.
+init 10 python:
+    import time
+    global mas_afk_time
+    global mas_afk_duration
+    global mas_afk_already
+    mas_afk_already = False
+    # How much time in seconds should pass, before it's considered that player is afk.
+    mas_afk_duration = 600
+    mas_afk_time = time.time() + mas_afk_duration
+    config.overlay_screens.append("inactivity_detection_screen")
+    class MASInactivity(renpy.Displayable):
+        
+        import pygame
+
+        def __init__(self):
+            super(MASInactivity, self).__init__()
+
+        def event(self, ev, x, y, st):
+            global mas_afk_time
+            global mas_afk_duration
+            # Constant checking for player activity - mouse movement, keystrokes and mouse button clicks
+            if ev.type == pygame.MOUSEMOTION or ev.type == pygame.KEYDOWN or ev.type == pygame.MOUSEBUTTONDOWN:
+                mas_afk_time = time.time() + mas_afk_duration
+            if mas_afk_time < time.time():
+                mas_afk_time = time.time() + 10000000000
+                # Checking if event can be restarted.
+                if not mas_isRstBlk(persistent.current_monikatopic):
+                    pushEvent(persistent.current_monikatopic)
+                    pushEvent('continue_event')
+                    persistent.current_monikatopic = 0
```

Comment:
we have a function called `restartEvent` which literally does this.

Why do you need to call in the other case vs jump in this one?